content/essentials/intro-to-reactivity.mdx

```diff
@@ -0,0 +1,186 @@
+During this phase, signals are initialized with their respective starting values. 
+These initialized signals serve as the primary data sources that the rest of the system will rely on.
+
+### 2. Subscription
+
+Once signals are established, the next step involves the effects indicating their dependencies. 
+In this phase, effects "subscribe" to specific signals, marking their interest and establishing a link. 
+This subscription keeps the effects aware of any changes to the signals they are tied to.
+
+### 3. Notification
+
+During the notification phase, any update in a signal's value sets off a chain reaction. 
+The altered signal proactively notifies all its subscribed effects about the modification. 
+Upon receiving this notification, these effects then update or activate as necessary to keep the system's state current and synchronized with the latest data changes.
+
+## Synchronous vs. asynchronous
+
+Reactivity in systems can manifest primarily in two ways: synchronous and asynchronous. 
+The choice between these often depend on the requirements of the system and the nature of the tasks involved.
+
+### Synchronous reactivity
+Synchronous reactivity operates in a direct and linear fashion. 
+When a signal changes, its corresponding effects are immediately updated in a sequential manner. 
+This form of reactivity provides a direct correlation between a signal's change and the system's response, guaranteeing that the system remains consistently aligned with the most recent data.
+
+### Asynchronous reactivity
+Asynchronous reactivity can improve performance in some cases. 
+Unlike its synchronous counterpart, updates to effects are not instantaneous. 
+Instead, they may be delayed or could wait for the completion of an external event or task.
+
+This approach is useful in scenarios where an effect depends on multiple signals and when updating one signal before another may result a data inconsistency. 
+In such a scenario, [`batch`](/reference/batch) can be used to delay triggering effects so the effect runs after both signals have been updated.
+
+## Efficient data management
+
+### Memo
+
+Memo, short for memoization, is a technique that is used in reactive systems to improve performance.
+In these systems, memos and effects share some similarities, with the primary difference being that memos *return* a signal as their return value.
+Furthermore, like effects, memos use dependency tracking  to make sure that recalculations and updates are triggered only when changes occur within their dependent signals.
+
+The value of a memo lies in its ability to selectively reclaculate; this happens only when there are changes in the dependent signals.
+This process involves **caching** the results from resource-intensive function calls.
+Chaching stores the results of previous calculations, allowing the system to return a cached result rather than reclaculating when the same inputs are encountered.
```

Comment:
```suggestion
Caching stores the results of previous calculations, allowing the system to return a cached result rather than recalculating when the same inputs are encountered.
```